modules/utxo-lib/src/bitgo/Musig2.ts

```diff
@@ -0,0 +1,214 @@
+import { PSBT_PROPRIETARY_IDENTIFIER, PsbtProprietaryKeyValueData, UtxoPsbt } from './UtxoPsbt';
+import { checkPlainPublicKey, checkXOnlyPublicKey, toXOnlyPublicKey } from './outputScripts';
+import { BIP32Interface } from 'bip32';
+import { ecc, musig } from '../noble_ecc';
+import { Tuple } from './types';
+import { tapTweakPubkey } from '../taproot';
+import { TapBip32Derivation } from 'bip174/src/lib/interfaces';
+
+export const PSBT_IN_MUSIG2_PARTICIPANT_PUB_KEYS_SUBTYPE = 0x01;
+export const PSBT_IN_MUSIG2_PUB_NONCE_SUBTYPE = 0x02;
+
+/**
+ *  Participant pub key value object.
+ */
+export interface PsbtMusig2ParticipantsKeyValueData {
+  keyData: {
+    tapOutputKey: Buffer;
+    tapInternalKey: Buffer;
+  };
+  valueData: {
+    participantPubKeys: Tuple<Buffer>;
+  };
+}
+
+/**
+ * Psbt proprietary key val util function for participants pub keys. SubType is 0x01
+ * Ref: https://gist.github.com/sanket1729/4b525c6049f4d9e034d27368c49f28a6
+ * @return x-only tapOutputKey||tapInternalKey as sub keydata, plain sigining participant keys as valuedata
+ */
+export function encodePsbtMusig2ParticipantsKeyValData(
```

Comment:
Seems like it would make the most sense to either have this function take `PsbtMusig2ParticipantsKeyValueData` or have the decode function return the same types as his encode function. (Personally, I'd change `PsbtMusig2ParticipantsKeyValueData` to
```
tapOutputKey: Buffer;
tapInternalKey: Buffer;
participantPubKeys: [Buffer, Buffer];
```
 and then use that in both places.

Reply:
Done